test/spmd/tensor/test_tp_sharding_ops.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+from spmd.tensor.placement_types import Replicate
+import torch
+from torch.testing._internal.common_utils import run_tests
+from ..test_utils import DistTensorTestBase, with_comms, TEST_GPU_NUM
+from spmd import DeviceMesh, Tensor, Shard, Replicate, distribute_tensor
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+
+
+class TPShardingOpsTest(DistTensorTestBase):
+    # TODO: We need to add CPU tests for ops in the future.
+    @with_comms
+    @skip_if_lt_x_gpu(TEST_GPU_NUM)
+    def test_sharded_view(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        torch.manual_seed(0)
+        tensor = torch.rand(16, 35, 26).cuda(self.rank)
```

Comment:
we don't need to move tensor to cuda, `distributed_tensor` or `from_local` automatically do this for us